data/recommendation/notify.js

```diff
@@ -1,68 +1,140 @@
+const yo = require('yo-yo');
+
 class Notify {
   start() {
-    this.setup();
+    this.target = this.createBase();
+    document.body.appendChild(this.target);
+    const methodsToBind = ['handleEndWarning', 'handleEndSignup'];
+    for (let key of methodsToBind) { // eslint-disable-line prefer-const
+      this[key] = this[key].bind(this);
+    }
     self.port.on('data', recs => {
-      this.showAdvice(recs);
+      const date = new Date(recs.BreachDate);
+      const year = date.getFullYear();
+      this.render({
+        warn: true,
+        site: recs.Title,
+        year,
+      });
     });
+    this.hasSync = false;
     self.port.on('syncEnabled', () => {
-      this.removeSyncSignup();
+      this.hasSync = true;
     });
     self.port.on('syncDisabled', () => {
-      this.showSyncSignup();
+      this.hasSync = false;
```

Comment:
I don't think it's necessary to re-render. Any user interaction outside the panel, like signing up for sync, will cause the panel to automatically close. Based on that, I believe that the 'syncDisabled' signal will never emit while the panel is open, and re-render won't be used. 

Also, the render function never uses `hasSync`. Rather, the boolean is used to decide if users should be offered a sync account. That decision results in the render function being called to offer sync, or the panel automatically closing.